src/app-bar.jsx

```diff
@@ -103,6 +114,26 @@ const AppBar = React.createClass({
     titleStyle: React.PropTypes.object,
 
     /**
+     * Settings object for position waterfall. Should at least have minHeight
+     * and maxHeight properties, both numeric. These specify min and max visual heigth
+     * of the component while window scrolling. Optional children property can be a node
+     * or a function (will receive component styles object as argument) returning a node. This node will
+     * be inserted in the slide (scrolled) element of the component. Optional onHeightChange property
+     * is a function called when visual height of the component changes on scroll. This function will
+     * receive as arguments an object with height, minHeight, maxHeight and childrenEl (DOM element of
+     * the component) properties. Using onHeightChange, animation effects can be achieved
+     * by altering style properties of specific DOM elements.
+     */
+    waterfall: React.PropTypes.shape({
```

Comment:
I can do this, but please let me know if it's the only problem left. I don't want to keep changing things and then get anyway a reject because of another bigger things.